### DISCUSSION TITLE
File upload and download

### DISCUSSION THREAD

**Discussion by guzman-raphael** [community/community#47531] [Codespaces]
### Select Topic Area

Product Feedback

### Body

First, I'd like to stay that I am loving GitHub Codespaces! A big thank you to the team behind this. :clap: 

Codespaces is already making a big difference in how my team approaches team development. We've now started thinking of possibly using it as a platform for delivering educational workshops. We run several throughout the year for open source projects around Neuroscience / Data Science.

In validating and preparing for that, one key piece is to share some training data and allow users to bring their own data to the workshop. This is usually in the realm of <=5GB which the free tier should easily accommodate.

However, I started running into some issues testing the experience of uploading and downloading using the VSCode Browser UI. The primary issue was around the slow speeds I was getting. That inspired me to record some tests and the following are the results. These were all tested in sequence on the same network. Tested from home while I was the only one connected.

## Upload

Platform | Source | Destination | Size [GB] | Duration [m] | Bandwidth [Mbps] | Note
-- | -- | -- | -- | -- | -- | --
Ookla speedtest | user machine | internet | --- | --- | 317 |  
Google speedtest | user machine | internet | --- | --- | 312 |  
AT&T speedtest | user machine | internet | --- | --- | 328 |  
Google Drive | user machine | Google Drive | 5 | 4 | 170.6666667 |  
GH Codespaces | GH Codespaces | S3 | 5 | 1.137777778 | 600 |  
GH Codespaces | user machine | GH Codespaces | 5 | 20 | 34.13333333 |  

![Screenshot from 2023-02-14 10-24-22](https://user-images.githubusercontent.com/38401847/218999402-e5cf6a01-d826-4688-82de-9bf6111bdb7a.png)

## Download

Platform | Destination | Source | Size [GB] | Duration [m] | Bandwidth [Mbps] | Note
-- | -- | -- | -- | -- | -- | --
Ookla speedtest | user machine | internet | --- | --- | 367 |  
Google speedtest | user machine | internet | --- | --- | 361 |  
AT&T speedtest | user machine | internet | --- | --- | 344 |  
Google Drive | user machine | Google Drive | 5 | 2 | 341.3333333 |  
GH Codespaces | GH Codespaces | S3 | 5 | 0.6826666667 | 1000 |  
GH Codespaces | user machine | GH Codespaces | 5 |   | #DIV/0! | Did not finish

![Screenshot from 2023-02-14 10-50-43](https://user-images.githubusercontent.com/38401847/218999343-4a8f6708-3d28-4b3f-98a7-b82a4e0e9b63.png)

As you can see from the last result, it did not finish downloading the data set at all.

## Questions

My burning questions from this exercise are:
- Do know why my upload/download speeds between Codespaces would be so different than other similar tests between a user machine and the internet?
- Any idea why the download fails?
- Is there any configuration or tuning I can do to improve this from my side?

## Additional Details

In case it helps, here's some more info on my setup:
- Codespace
  - Machine Type: 2-core • 4GB RAM • 32GB storage
  - Region: US-East
  - Tested by performing a right-click `Upload` or `Download` on the target in the VSCode Browser tree
- User machine
  - Linux, Pop!_OS 22.04 LTS
  - Browser: SideKick Web Browser, Version 106.33.1.28262 (Official Build) stable (64-bit)
  - Similar results in Google Chrome as well (Version 110.0.5481.77 (Official Build) (64-bit))

**Comment by emisparkles** [comment]
Hi @guzman-raphael, thank you for your feedback! I'm glad to hear you are having a great experience with Codespaces.

I have reached out to other team members to see if we can answer your questions. In the meantime, you can use the GitHub CLI to speed up your uploads and downloads. You can run `gh codespace cp` in a local terminal to copy files between your local machine and the remote codespace. See the [documentation](https://cli.github.com/manual/gh_codespace_cp) for more details.

Thanks again for using Codespaces!

**Reply by guzman-raphael** [reply]
@emisargent Np and thanks for the quick response!

I'll give your suggestion a try using the `gh` CLI. Will post an update once I have results.
